airbyte-integrations/bases/source-acceptance-tests/src/main/java/io/airbyte/integrations/acceptance_tests/source/FileBasedSourceAcceptanceTestDriver.java

```diff
@@ -0,0 +1,160 @@
+      };
+      testConfigs.add(config);
+    }
+    return testConfigs;
+  }
+
+  IncrementalAcceptanceTestRunner.Config readIncrementalInputs(String connectorImage, IncrementalConfig config) {
+    JsonNode connectorConfig = deserializeJsonAtPath(config.getConnectorConfigPath(), "connector config");
+    var configuredCatalog = deserializeConfiguredCatalogAtPath(config.getConfiguredCatalogPath());
+    return new IncrementalAcceptanceTestRunner.Config(connectorImage, connectorConfig, configuredCatalog);
+  }
+
+  FullRefreshAcceptanceTestRunner.Config readFullRefreshInputs(String connectorImage, FullRefreshConfig config) {
+    JsonNode connectorConfig = deserializeJsonAtPath(config.getConnectorConfigPath(), "connector config");
+    var configuredCatalog = deserializeConfiguredCatalogAtPath(config.getConfiguredCatalogPath());
+    return new FullRefreshAcceptanceTestRunner.Config(connectorImage, connectorConfig, configuredCatalog);
+  }
+
+  CoreAcceptanceTestRunner.Config readCoreInputs(String connectorImage, CoreConfig coreConfig) {
+    var catalog = deserializeConfiguredCatalogAtPath(coreConfig.getConfiguredCatalogPath());
+    JsonNode expectedSpec = deserializeJsonAtPath(coreConfig.getSpecPath(), "specification");
+    JsonNode validConnectorConfig = deserializeJsonAtPath(coreConfig.getValidConnectorConfigPath(), "valid connector config");
+    JsonNode invalidConnectorConfig = deserializeJsonAtPath(coreConfig.getInvalidConnectorConfigPath(), "invalid connector config");
+
+    return new CoreAcceptanceTestRunner.Config(connectorImage, expectedSpec, validConnectorConfig, invalidConnectorConfig, catalog);
+  }
+
+  private SourceAcceptanceTestInputs parseArguments(String[] args) throws IOException {
+    ArgumentParser parser = ArgumentParsers.newFor(FileBasedSourceAcceptanceTestDriver.class.getName()).build()
+        .defaultHelp(true)
+        .description("Run source acceptance tests");
+
+    parser.addArgument("--testConfig")
+        .required(true)
+        .help("Path to the source acceptance test input configuration");
+
+    Namespace ns;
+    try {
+      ns = parser.parseArgs(args);
+    } catch (ArgumentParserException e) {
+      throw new IllegalArgumentException(e);
+    }
+
+    String testConfigPath = ns.getString("testConfig");
+    String testConfig = Files.readString(Path.of(testConfigPath));
+    return Yamls.deserialize(testConfig, SourceAcceptanceTestInputs.class);
+  }
+
+  void runTestSuite(String[] args) throws IOException {
+    SourceAcceptanceTestInputs testInputs = parseArguments(args);
+    List<Object> testConfigs = readTestInputs(testInputs);
+    for (Object testConfig : testConfigs) {
+      Class<?> klass = testConfig.getClass();
+
+      if (klass == IncrementalAcceptanceTestRunner.Config.class) {
+        IncrementalAcceptanceTestRunner.CONFIG = (IncrementalAcceptanceTestRunner.Config) testConfig;
```

Comment:
I'm struggling to understand how the `X.Config` set here are translated into `TestPlan`s in the `TestRunner` class. Is there something I can read to understand this?

Reply:
i think what's happening is he is trying to pass information to a junit test suite before it runs at all. so the way we get around doing this is by setting it as a static field on the class. i think when he implements the test classes themselves it'll make a bit more sense.